packages/fluentui/styles/src/mergeThemes.ts

```diff
@@ -304,12 +348,46 @@ export const mergeThemeVariables =
  * Component styles adhere to the same pattern as component variables, except
  *   that they return style objects.
  */
-export const mergeThemeStyles = (
-  ...sources: (ThemeComponentStylesInput | null | undefined)[]
+const mergeThemeStyles__PROD = (
+  themeComponentStylesA: ThemeComponentStylesInput | undefined,
+  themeComponentStylesB: ThemeComponentStylesInput | undefined,
 ): ThemeComponentStylesPrepared => {
+  if (themeComponentStylesA && themeComponentStylesB) {
+    const displayNames = _.union(..._.map([themeComponentStylesA, themeComponentStylesB], _.keys));
+
+    return displayNames.reduce((themeComponentStyles, displayName) => {
+      themeComponentStyles[displayName] = mergeComponentStyles(
+        themeComponentStylesA[displayName],
+        themeComponentStylesB[displayName],
+      );
+
+      return themeComponentStyles;
+    }, {});
+  }
+
+  if (themeComponentStylesA) {
+    return Object.fromEntries(
+      Object.entries(themeComponentStylesA).map(([displayName, styles]) => {
+        return [displayName, mergeComponentStyles(undefined, styles)];
+      }),
+    );
+  }
+
+  if (themeComponentStylesB) {
+    return Object.fromEntries(
+      Object.entries(themeComponentStylesB).map(([displayName, styles]) => {
+        return [displayName, mergeComponentStyles(undefined, styles)];
+      }),
+    );
+  }
+
+  return {};
+};
+
+const mergeThemeStyles__DEV: typeof mergeThemeStyles__PROD = (componentStylesA, componentStylesB) => {
   const initial: ThemeComponentStylesPrepared = {};
 
-  return sources.reduce<ThemeComponentStylesPrepared>((themeComponentStyles, next) => {
+  return [componentStylesA, componentStylesB].reduce<ThemeComponentStylesPrepared>((themeComponentStyles, next) => {
```

Comment:
All other `__DEV` functions call `_PROD` unless debug is enabled. Shouldn't this one do the same?

Reply:
Makes sense, updated.